ownership-chain/pallets/block-rewards-handler/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+	pallet_prelude::*,
+	traits::{Currency, ExistenceRequirement},
+};
+
+use sp_runtime::{traits::Zero, ArithmeticError};
+
+mod benchmarking;
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use super::*;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	pub type BalanceOf<T> =
+		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+
+		/// The currency type
+		type Currency: Currency<Self::AccountId>;
+	}
+
+	/// Source of rewards for block producers
+	#[pallet::storage]
+	#[pallet::getter(fn rewards_account)]
+	pub type RewardsAccount<T: Config> = StorageValue<_, T::AccountId, OptionQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub rewards_account: Option<T::AccountId>,
+	}
+
+	impl<T: Config> Default for GenesisConfig<T> {
```

Comment:
sorry, just saw this, `Default` can be derived here as well. Maybe you will need `DefaultNoBound` or sth like that